test/common/tcp/conn_pool_test.cc

```diff
@@ -403,6 +421,110 @@ TEST_F(TcpConnPoolImplTest, DrainConnections) {
   EXPECT_TRUE(conn_pool_->isIdle());
 }
 
+/**
```

Comment:
Is it possible to get an e2e test as well?  If there's no existing e2e tests with idle connection reuse we could configure it so we reclaim eagerly (1s or less), turn up connection preconnect so that a downstream request kicks off 2 upstream connections and then wait on the stat for the upstream connection to be reclaimed?

Reply:
Does e2e in the context mean integration test? We did test in our environment for days.

Reply:
e2e means /test/integration/tcp_proxy_integration_test.cc :-)  I'm glad you tested internally (that's actually pretty awesome) but it'd be good to have it also end to end regression tested.
I think if you set your new configuration option to a short timeout you can adapt one of the tests which does Preconnect and verify the connection gets reclaimed by waiting on cluster.cluster_0.upstream_cx_destroy